src/PostgREST/Middleware.hs

```diff
@@ -103,7 +102,6 @@ pgrstMiddleware logLevel =
     logger
   . gzip def
   . cors corsPolicy
-  . staticPolicy (only [("favicon.ico", "static/favicon.ico")])
```

Comment:
Hmm.. wait, I think this is not unreachable. If it's removed it will lead to 404 errors when doing a request through chrome:

I went to the address bar with: http://localhost:3000/projects

And the logs say:

```
127.0.0.1 - - [04/Jan/2021:11:28:08 -0500] "GET /projects HTTP/1.1" 200 - "" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.87 S
afari/537.36"
127.0.0.1 - - [04/Jan/2021:11:28:08 -0500] "GET /favicon.ico HTTP/1.1" 404 - "http://localhost:3000/projects" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML,
like Gecko) Chrome/78.0.3904.87 Safari/537.36"
```

Reply:
Hm. I had a feeling that my "test" was not a 100%.

I tried to access `/favicon.ico` with Firefox but got a 404 (with the code still in it). Maybe I didn't use the right accept headers or something?

Reply:
You can reproduce by doing a hard refresh(ctrl + shift + r) on chrome.

Reply:
Ah.. I've just noticed that this also happens on 7.0.1, so the `staticPolicy` is not doing anything.

Reply:
I just repeated that test... and the favicon loaded fine...

I probably tested the first time... when I had already removed it :D

I will add it back in and add a basic IO test instead.

I wonder, however: Are we packaging that file in the Docker container, too??

Reply:
> Ah.. I've just noticed that this also happens on 7.0.1, so the `staticPolicy` is not doing anything.

Wait, now I'm confused. It seemed to work for me on the current main branch, now.

Reply:
Actually... I might have tested the favicon on the docker container before removing it. Maybe that's the real issue here.

Reply:
> Ah.. I've just noticed that this also happens on 7.0.1, so the staticPolicy is not doing anything.

Confused as well, I ran the binary I have installed and the favicon is 404ing. Maybe the favicon.ico is not being packaged into the static binary? Not sure..

Reply:
Yes, I was only able to make it work by running it with `postgrest-run`. All packages are missing it, because it's only read at run-time right now.

Reply:
True, I also get the favicon with postgrest-run. Then that's another issue.


(Btw, the favicon is empty. Perhaps we should replace it with the [docs favicon](https://raw.githubusercontent.com/PostgREST/postgrest-docs/master/_static/favicon.ico))

Reply:
I see two ways to go about this. We should either:
- serve the docs favicon as static file embedded into the executable
- or just remove the whole thing and add a condition in the logger to never log favicon errors

Given that the icon is empty currently, it seems like the reason to put this in was just about logging?

Reply:
> or just remove the whole thing and add a condition in the logger to never log favicon errors

Though it would still show as an error on chrome console.

> serve the docs favicon as static file embedded into the executable

Would this be hard to do?

Reply:
I found this:
https://hackage.haskell.org/package/wai-middleware-static-embedded

I will give it a try.

Reply:
After the discussion in #2092 I am not sure anymore, whether we should add back the favicon. It does "pollute" the api namespace.

And I'm not sure about adding a condition to the logger either. If somebody used this endpoint somehow, it would be really strange to not log requests about it.

I'll just remove the dysfunctional code and stick with that for now.

Reply:
> It does "pollute" the api namespace.

True. Just realized this as well.

> If it's removed it will lead to 404 errors when doing a request through chrome
> Ah.. I've just noticed that this also happens on 7.0.1

Since nobody complained maybe the 404 was a non-issue from the get go. Removing it should be fine.